t_modify_http_headers/test_logic.py

```diff
@@ -0,0 +1,355 @@
+__copyright__ = "Copyright (C) 2023 Tempesta Technologies, Inc."
+__license__ = "GPL2"
+
+from framework import tester
+from helpers.deproxy import H2Response, HttpMessage, Request, Response
+
+
+def generate_http1_request(optional_headers=[]) -> str:
+    return (
+        f"GET / HTTP/1.1\r\n"
+        + "Host: localhost\r\n"
+        + "Connection: keep-alive\r\n"
+        + "".join(f"{header[0]}: {header[1]}\r\n" for header in optional_headers)
+        + "\r\n"
+    )
+
+
+def generate_response(optional_headers=[]) -> str:
+    return (
+        "HTTP/1.1 200 OK\r\n"
+        + f"Date: {HttpMessage.date_time_string()}\r\n"
+        + "Server: Tempesta FW/pre-0.7.0\r\n"
+        + "".join(f"{header[0]}: {header[1]}\r\n" for header in optional_headers)
+        + "Content-Length: 0\r\n\r\n"
+    )
+
+
+def generate_h2_request(optional_headers=[]) -> list:
+    return [
+        (":authority", "localhost"),
+        (":path", "/"),
+        (":scheme", "https"),
+        (":method", "GET"),
+    ] + optional_headers
+
+
+class TestLogicBase(tester.TempestaTest, base=True):
+    tempesta = {
+        "config": """
+    listen 80;
+    listen 443 proto=h2;
+    
+    server ${server_ip}:8000;
+
+    tls_certificate ${tempesta_workdir}/tempesta.crt;
+    tls_certificate_key ${tempesta_workdir}/tempesta.key;
+    tls_match_any_server_name;
+    """,
+    }
+
+    clients = [
+        {
+            "id": "deproxy-1",
+            "type": "deproxy",
+            "addr": "${tempesta_ip}",
+            "port": "80",
+        },
+    ]
+
+    backends = [
+        {
+            "id": "deproxy",
+            "type": "deproxy",
+            "port": "8000",
+            "response": "static",
+            "response_content": "",
+        }
+    ]
+
+    cache: bool
+    requests_n: int
+    directive: str
+    h2: bool
+
+    def update_tempesta_config(self, config: str):
+        tempesta = self.get_tempesta()
+        cache = "cache 2;\ncache_fulfill * *;\n" if self.cache else "cache 0;"
+        tempesta.config.defconfig += config + "\n" + cache
+
+    def base_scenario(self, config: str, optional_headers: list, expected_headers: list):
+        self.update_tempesta_config(config=config)
+        self.start_all_services()
+
+        server = self.get_server("deproxy")
+        client = self.get_client("deproxy-1")
+
+        server.set_response(generate_response(optional_headers))
+
+        # send 2 requests for checking cache and dynamic table
+        for _ in range(2 if self.cache else self.requests_n):
+            client.send_request(
+                generate_h2_request(optional_headers)
+                if self.h2
+                else generate_http1_request(optional_headers),
+                "200",
+            )
+
+        expected_response = self.check_response(optional_headers, expected_headers, client)
+        expected_request = self.get_expected_request(optional_headers, expected_headers, client)
```

Comment:
`check_response` and `get_expected_request` do same things but have different names.

Reply:
Done. It's typo.